crates/store/src/backend/azure/mod.rs

```diff
@@ -126,6 +126,7 @@ impl AzureStore {
                             }
                             Err(e) => {
                                 err = Some(e);
+                                break;
```

Comment:
This was something I noticed after I submitted my previous PR; if we get an error, we don't have to keep consuming from the stream, we can just break out right away and return the error to the caller.

The code would be correct either way, but it seems more efficient to do it this way and break out as soon as we know for sure we have an error.